fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/pattern/Pattern.java

```diff
@@ -69,6 +85,10 @@ public boolean matchWithoutChild(GroupExpression expression) {
             return true;
         }
 
+        if (isPatternScan() && scanTypes.contains(expression.getOp().getOpType())) {
```

Comment:
why need to change here?

Reply:
I add a PATTARN_SCAN, here I should judge whether current op matches PATTERN_SCAN.
or anywhere else I can put this logic?